### PR TITLE
[PM-10010] Move close/cancel buttons to upper-left

### PR DESCRIPTION
**Pull request by KatherineInCode**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10010

## 📔 Objective

One point of feedback from early testers is that the iOS app changed the order of buttons in the top bar from the MAUI app. This updates the various sheets and screens throughout the app to match MAUI, particularly in that Close or Cancel buttons are in the top-left and action buttons are in the top-right.

## 📸 Screenshots

See diff of screenshot tests.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: BitwardenShared/UI/Tools/Generator/Generator/GeneratorView.swift [hunk 1]
```diff
@@ -48,19 +48,21 @@ struct GeneratorView: View {
...
         .toolbar {
-            ToolbarItemGroup(placement: .topBarTrailing) {
-                optionsToolbarMenu {
-                    Button(Localizations.passwordHistory) {
-                        store.send(.showPasswordHistory)
-                    }
-                }
-
+            ToolbarItem(placement: .topBarLeading) {
                 if store.state.presentationMode.isDismissButtonVisible {
```
Comment (phil-livefront) [inline BitwardenShared/UI/Tools/Generator/Generator/GeneratorView.swift:52]: 🤔 should creating the `ToolbarItem` go below the `if` condition so its only created if the presentation mode allows a dismiss button?

File: BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewItemView.swift [hunk 2]
```diff
@@ -46,7 +46,26 @@ struct ViewItemView: View {
...
+                closeToolbarButton {
+                    store.send(.dismissPressed)
+                }
+            }
+
             ToolbarItemGroup(placement: .navigationBarTrailing) {
+                if let state = store.state.loadingState.data {
+                    if state.isSoftDeleted {
+                        toolbarButton(Localizations.restore) {
+                            await store.perform(.restorePressed)
```
Comment (phil-livefront) [inline BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewItemView.swift:59]: 🤔 should we add ViewInspector tests for these?

**Review by phil-livefront — COMMENTED** [review]
Looks good!  A couple questions but from the snapshot changes this looks great.

File: BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewItemView.swift (hunk 2 above)
Comment (KatherineInCode) [reply BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewItemView.swift:59]: I spent a fair bit of time fiddling with the data to get into the "deleted item" branch, and it's just creating an entire second set of data to get a CipherView that's set as deleted. Should be very doable, just would vastly increase the size of the file, which is I assume the reason it wasn't done before.

Comment (phil-livefront) [reply BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewItemView.swift:59]: ah yeah, that makes sense to me!

File: BitwardenShared/UI/Tools/Generator/Generator/GeneratorView.swift (hunk 1 above)
Comment (KatherineInCode) [reply BitwardenShared/UI/Tools/Generator/Generator/GeneratorView.swift:52]: SwiftUI doesn't like putting the `if` higher, as "'buildIf' is only available in application extensions for iOS 16.0 or newer".

Comment (phil-livefront) [reply BitwardenShared/UI/Tools/Generator/Generator/GeneratorView.swift:52]: you're right, silly me.  My last project was iOS 17 based so I was a bit spoiled in this sense.  thanks for the reminder!

**Review by phil-livefront — APPROVED** [review]

File: BitwardenShared/UI/Tools/Send/SendItem/AddEditSendItem/AddEditSendItemView.swift [hunk 3]
```diff
@@ -77,6 +77,10 @@ struct AddEditSendItemView: View { // swiftlint:disable:this type_body_length
                         )
                     )
                 }
+
+                cancelToolbarButton {
+                    store.send(.dismissPressed)
+                }
```
Comment (fedemkr) [inline BitwardenShared/UI/Tools/Send/SendItem/AddEditSendItem/AddEditSendItemView.swift:83]: 🤔 Is it ok for the cancel button to be to the right of the avatar in the share extension? I feel like the cancel button should be the first thing on the toolbar to the left; like in `VaultAutofillListView`.
| (Cancel) (Avatar)     Title     (Other actions) |

Comment (KatherineInCode) [reply BitwardenShared/UI/Tools/Send/SendItem/AddEditSendItem/AddEditSendItemView.swift:83]: Good catch! I thought I found all of those, but I guess not!

I guess we don't have any snapshot tests of the share extension, which is how I missed this, I think. I'll swing back around and maybe try to get some more snapshots in.

Comment (fedemkr) [reply BitwardenShared/UI/Tools/Send/SendItem/AddEditSendItem/AddEditSendItemView.swift:83]: Awesome thanks!

**Review by matt-livefront — DISMISSED** [review]

**Pull request merged**